### DISCUSSION TITLE
Show max characters on input and limit

### DISCUSSION THREAD

**Discussion by jekuer** [directus/directus#2994] [Feature Requests]
With V8, we had a beautiful option to show a character count to the user.
It would be an important feature to limit the input of a specific text field.

**Long shot:**

1. An additional numeric input at the field settings which holds an optional char limit.
2. If set (or if the field is defined as String with a max size), we show a small character count "X/200" and limit the input (user cannot add more than those 200 characters).
3. Show the count only if we have >0 chars as value.

**Quick shot:**

1. We apply this only to input fields, which are set as String and therefore already hold a "natural" low limit. Since we can already edit this (even if it is more a database thing), it can act as an indirect char limit.
2. For this type of field, we use the UI logic from above on input.
3. This can be extended later by adding a more explicit option (see long shot).

Relates to https://github.com/directus/next/issues/1009
(I hope this is no duplicate)

**Comment by benhaynes** [comment]
I'm not sure I follow your "long shot" use case, but I am down to re-implement the same feature we had in v8, in v9. Basically we would use the column length value (eg `200` chars) and would show the chars remaining as you type. I think this would be an Interface Option, and maybe off by default?

**Reply by rijkvanzanten** [reply]
We already have the `length` under schema when setting up a `string` (eg VARCHAR) type field. 

Once you get in the final 20% of the text-input length, it'll show this:

![CleanShot 2020-11-18 at 12 37 54](https://user-images.githubusercontent.com/9141017/99566455-ed6d5580-299a-11eb-92d1-12e3d6c60161.gif)

**Reply by rijkvanzanten** [reply]
Now, @jekuer there is a bug where this length isn't read correctly, so i'll push up a fix for that rn 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
